### PR TITLE
private repo to content dependencies check

### PR DESCRIPTION
**Pull request by jochman**
find-dependencies will merge the dependencies from local and content repo.

fixes: https://github.com/demisto/etc/issues/33848
image
image

### CODE REVIEW

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -21,22 +21,23 @@
 import requests
 import urllib3
 import yaml
+# disable insecure warnings
```
Comment (roysagi) [inline demisto_sdk/commands/common/tools.py:24]: why did you move this comment?

File: demisto_sdk/__main__.py [hunk 2]
```diff
@@ -911,11 +911,9 @@ def id_set_command(**kwargs):
 
 
 # ====================== merge-id-sets ====================== #
-@main.command(name='merge-id-sets',
-              hidden=True,
-              short_help='Merge two id_sets')
-@click.help_option(
-    '-h', '--help'
```
Comment (roysagi) [inline demisto_sdk/__main__.py:918]: why remove the help?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 3]
```diff
@@ -1151,21 +1178,28 @@ def build_dependency_graph(pack_id: str, id_set: dict, verbose: bool,
                 leaf_dependencies = PackDependencies._find_pack_dependencies(
                     leaf, id_set, verbose=verbose, exclude_ignored_dependencies=exclude_ignored_dependencies)
 
-                if leaf_dependencies:
```
Comment (roysagi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:1154]: I think this handles `leaf_dependencies=None` which was an issue in the past.

**Review by roysagi — CHANGES_REQUESTED** [review]

File: demisto_sdk/__main__.py (hunk 2 above)
Comment (jochman) [reply demisto_sdk/__main__.py:918]: Because it is uneeded. It will use global help.
What going on here is adding -h as help. --help will always work.

File: demisto_sdk/commands/common/tools.py (hunk 1 above)
Comment (jochman) [reply demisto_sdk/commands/common/tools.py:24]: by mistake. I've removed it

**Comment by jochman** [comment]
sdk nightly and upload trigger builds: https://app.circleci.com/pipelines/github/demisto/content?branch=test-new-deps

**Review by roysagi — APPROVED** [review]
